ghcide/src/Development/IDE/Core/ProgressReporting.hs

```diff
@@ -143,7 +157,8 @@ delayedProgressReporting before after lspEnv optProgressStyle = do
                 loop _ _ | optProgressStyle == NoProgress =
                     forever $ liftIO $ threadDelay maxBound
                 loop id prev = do
-                    InProgress{..} <- liftIO $ readVar inProgress
+                    done <- liftIO $ readTVarIO doneVar
+                    todo <- liftIO $ readTVarIO todoVar
```

Comment:
Should these be in an atomic transaction? what happens if someone updates `todo` between the two lines?

Reply:
Atomicity is not crucial here as the values are only used to display a progress report and it's ok to be off by one.